Fix realsat ids

hcorson-dosch opened the pull request:
Noticed during coding the munging of the realsat data w/ Rasha that Pepacton had very few data points (14) compared to Cannonsville, which didn't line up with what was [on the website](https://github.com/USGS-R/lake-temperature-model-prep/issues/199):
![image](https://user-images.githubusercontent.com/54007288/114245304-d7662e80-9955-11eb-8e4d-fe7b9c14b1cd.png)

I downloaded the base shapefile (`ReaLSAT.shp`) from http://umnlcc.cs.umn.edu/realsat/ and loaded it in ArcMap to check - looks like the id that was being used for Cannonsville actually corresponds to Pepacton, and the id for Pepacton was actually for a different reservoir.

Based on the `ReaLSAT.shp` shapefile, I pulled these ids for the two reservoirs:  Cannonsville: 573536, Pepacton: 573567. There is no attribute data other than id and continent in the base shapefile, so you have to go off of reservoir shape and location:
![image](https://user-images.githubusercontent.com/54007288/114245661-86a30580-9956-11eb-9d2a-9c1501df2722.png)

I updated the `realsat_ids` object in `2_observations.yml` to reflect these new ids, and also modified the munging script to convert the area to units of m2 to match the hypsometry data we're providing as input to GLM

I also went ahead and added the resulting csv `2_observations/out/realsat_monthly_surface_area.csv` to the `.gitignore` so that it would get committed, b/c of how much of a pain it is to download the ~2gb master zip files that contain the zipped shapefiles from http://umnlcc.cs.umn.edu/realsat/. 

After I'd decided to go ahead and do that Alison suggested we could upload to a shared cache instead, so I can modify that as needed.

Good news is - we now have lots more data for Pepacton!

aappling-usgs commented on 2_observations/src/munge_surface_area_data.R:
```diff
@@ -36,15 +36,17 @@ combine_realsat_reservoir_data <- function(reservoir_ids, out_file){
   })
   # using the reservoir_shapefiles (shape file paths).
   # read reservoirs shape file and convert it to dataframe.
+  # convert area (in units of LANDSAT 30-m pixels) to m2
```
Ooh, another good catch

aappling-usgs reviewed (changes requested):
Hi Hayley, lots of good changes in here. In additional to the file size / git conventions we discussed last week, one other thing about git committing realsat_monthly_surface_area.csv is that simply committing that file will not persuade `scipiper` on other computers that the file is up to date. So if I pull this code and call `scmake('2_observations/out/realsat_monthly_surface_area.csv')`, my computer will still complain that the file is out of date (and tell me "You must download reservoir monthly shapefile data"). That's temporarily workable because I am confident based on this PR and my conversations with you that the git-versioned file is up to date for now, but it's not especially robust to changes in munge_surface_area_data.R. A shared-cache approach would make it so I don't need to download the shapefiles to really know whether the csv is up to date. Given that it's a pretty minor switch, I'm requesting that you shift from using git to using Drive to store the file before we close out this PR.

limnoliver commented:
Thanks for your attention to detail, Hayley. I also found these manually for Rasha, and not exactly sure where I went wrong, but I'm glad you spotted the difference between expected and retrieved data.

limnoliver commented:
Just went back to figure out what I did wrong. I had the `Grand_id`s of both reservoirs (Pepacton = 2192, Cannsonville = 1550) which I noticed was a field in the HydroLakes dataset, and the hydrolakes ID (`Hylak_id`) was what was being used in the ReaLSAT dataset. So, I filtered the HydroLakes dataset by `grand_id` and found the `Hylak_id` of both reservoirs (Pepacton = 9130, Cannonsville = 9131), so I just tripped up here in the filtering/reporting steps. And here I thought using IDs to filter vs manually finding the reservoirs would be safer....ha!

Anyway, by going through this again, I noticed that Cannsonville actually has TWO IDs in ReaLSAT: `556766` and `573536`. Note `556766` is small, so not sure if it's worth bringing in and summing or just ignoring? I didn't look at the ReaLSAT data to confirm if there was timeseries data for `556766`, but just mentioning here for completeness. 

```
realsat <- sf::st_read('ReaLSAT.shp')
cannon <- filter(realsat, Hylak_id %in% '9131')
ggplot(cannon) +
  geom_sf(aes(color = factor(ID)))
```
![image](https://user-images.githubusercontent.com/15788176/114416993-d6601780-9b76-11eb-9b35-24d5c0a8d583.png)

Here's that portion of the reservoir from google maps.
![image](https://user-images.githubusercontent.com/15788176/114416877-bc263980-9b76-11eb-9844-8a0028f36944.png)

hcorson-dosch commented:
Good catch, Sam! Seems like we might as well add this data, since it exists?

hcorson-dosch commented:
@limnoliver  - I added in that second realsat_id for Cannonsvile and summarized the data
@aappling-usgs - I switched to pushing the munged surface area to Drive - not 100% sure I did that correctly

aappling-usgs reviewed (changes requested):
Hi Hayley, the new .ind file you did create looks good to be, but to reap the benefits of the shared cache, you'll also need to make 2_observations/out/raw_realsat_reservoir_data.csv into an indicator file target. I've added comments to help with that transition, though I don't think any of those suggestions will be news to you.

hcorson-dosch commented on 2_observations/out/realsat_monthly_surface_area.csv:
```diff
@@ -0,0 +1,553 @@
+reservoir,date,area_m2
+Cannonsville,1984-06-15,15957900
```
hmmm I might have messed this up, b/c I remove the file from the .gitignore yesterday, before I deleted it 😟

I deleted it now and committed the deletion, but I can't commit any changes to the .gitignore b/c `!2_observations/out/realsat_monthly_surface_area.csv` had already been removed....

hcorson-dosch replied:
Should I add `!2_observations/out/realsat_monthly_surface_area.csv` back to the .gitignore, commit the file, delete it, commit the deletion, remove it from .gitignore, and commit the changes to .gitignore?

aappling-usgs replied:
Yes, that's the way I've been able to solve such issues in the past. It feels painful to go back and forth (though in reality it shouldn't take too long). On the bright side, the pain will likely help you remember how to avoid the issue more often in the future =)

hcorson-dosch replied:
Ok! I think we're good to go now

aappling-usgs reviewed (approved):
Looks good to me! I will merge now because we've already been through a lot of round of review and I think are in agreement that this PR is done.
